Выполнено задание 10-го спринта

eugexl opened the pull request.

Commit:
Выполнено задание 10-го спринта

DanilaG commented on ImageFeed/Constants.swift:
```diff
@@ -0,0 +1,36 @@
+
+import Foundation
+
+struct UnsplashData {
+    
+    static let accessKey = "5WmzEZFA0XIUqHfm0sRalwInRzk70w9YK6FSOTRn-nw"
+    static let secretKey = "VNQJRTqyR8xj02sWpQN68HBwLxAgQFmZBYkoMJE-Jh0"
+    
+    static let redirectURI = "urn:ietf:wg:oauth:2.0:oob"
+    
+    static let accessScope = "public+read_user+write_likes"
+    static let defaultBaseURL = URL(string: "https://api.unsplash.com")!
+    
+    static let authorizeURLString = "https://unsplash.com/oauth/authorize"
+    static let tokenRequestURLString = "https://unsplash.com/oauth/token"
+    
+    static let postMethod = "POST"
+}
+
+struct Segues {
+    
+    static let showWebView  = "ShowWebView"
+    static let showAuthVC   = "ShowAuthViewController"
+}
+
+struct VCs {
+    
+    static let splashVC = "SplashViewController"
+    static let tabBarVC = "TabBarViewController"
+}
```
🔥, что разнёс все константы в namespace-ы. Это помогает не запутаться + упрощает работу с автодополнением)

DanilaG commented on ImageFeed/Constants.swift:
```diff
@@ -0,0 +1,36 @@
+    static let redirectURI = "urn:ietf:wg:oauth:2.0:oob"
+    
+    static let accessScope = "public+read_user+write_likes"
+    static let defaultBaseURL = URL(string: "https://api.unsplash.com")!
```
Давай избавимся от force unwrap?) Так как их использование может приводить к падениям приложения

Commit:
Внесены корректировки в соответствии с замечаниями/рекомендациями ревьюера:
- Поправлен статус-бар экранов авторизации;
- Изменён способ преобразования JSON-переменных с CodingKeys на KeyDecodingStrategy
